peachjam/models/core_document_model.py

```diff
@@ -155,6 +160,18 @@ def clean(self):
         except ValueError:
             raise ValidationError("Invalid FRBR URI: " + frbr_uri)
 
+        self.assign_frbr_uri()
+
+        expression_frbr_uri = self.generate_expression_frbr_uri()
+        if not self.pk or (self.initial_expression_frbr_uri != expression_frbr_uri):
```

Comment:
Rather than stashing the initial front uri, you can put .exclude(pk=self.pk) at the end of the query to look for entries other than this one.

Reply:
Neat.